packages/vx-xychart/src/hooks/useScales.ts

```diff
@@ -0,0 +1,81 @@
+import { AxisScaleOutput } from '@vx/axis';
+import { ScaleConfig, NumberLike, createScale, ScaleInput } from '@vx/scale';
+import { extent as d3Extent } from 'd3-array';
+import { useMemo } from 'react';
+import DataRegistry from '../classes/DataRegistry';
+
+/** A hook for creating memoized x- and y-scales. */
+export default function useScales<
+  XScaleConfig extends ScaleConfig<AxisScaleOutput>,
+  YScaleConfig extends ScaleConfig<AxisScaleOutput>,
+  Datum = unknown
+>({
+  xScaleConfig,
+  yScaleConfig,
+  dataRegistry,
+  xRange,
+  yRange,
+}: {
+  xScaleConfig: XScaleConfig;
+  yScaleConfig: YScaleConfig;
+  dataRegistry: DataRegistry<XScaleConfig, YScaleConfig, Datum>;
+  xRange: [number, number];
+  yRange: [number, number];
+}) {
+  // pull out memoization keys that are less likely to change
+  const registryKeys = dataRegistry.keys();
+  const [xMin, xMax] = xRange;
+  const [yMin, yMax] = yRange;
+
+  return useMemo(() => {
```

Comment:
should this be broken into multiple `useMemo`? Otherwise `x` will also update when only `y` is updated